python/MDSplus/tree.py

```diff
@@ -489,7 +488,7 @@ def __enter__(self):
     def __del__(self):
         if not self.public:
             self.__exit__()
-            _TreeShr.TreeFreeDbid(self.ctx)
+            _TreeShr.TreeFreeDbid(self._ctx)
```

Comment:
Why are some `self.ctx` and others `self._ctx`?

Reply:
ctx is the public access to _ctx and may redirect to TreeCtx() is public is True. _ctx is the private field used if public is False .. if not self.public -> avoid double check